packages/puppeteer-core/src/common/bidi/Page.ts

```diff
@@ -384,12 +398,15 @@ export class Page extends PageBase {
     if (this.#closedDeferred.finished()) {
       return;
     }
-    this.#closedDeferred.resolve(new TargetCloseError('Page closed!'));
-    this.#networkManager.dispose();
 
     await this.#connection.send('browsingContext.close', {
       context: this.mainFrame()._id,
     });
```

Comment:
missing call to this.dispose()?

Reply:
I think it would not be possible to re-order it like this though: you cannot call dispose before sending browsingContext.close but it would be too later after calling browsingContext.close.